### ISSUE TITLE
eCarePlanner 3.2.1 - Goals still not displaying, getConceptDisplayString() needs to perform undefined check on 'code' before checking 'code.text'

### ISSUE THREAD

**Issue by mattStorer** [chronic-care/mcc-project#482]
Goals are still not displaying in eCarePlanner 3.2.1.  This issue is related to #479.

### Error: 
```
TypeError: Cannot read properties of undefined (reading 'text')
    at getConceptDisplayString (goal.util.ts:309:12)
    at transformToMccGoalSummary (goal.util.ts:319:20)
    at Array.map (<anonymous>)
    at _callee3$ (goal.ts:160:55)
    at tryCatch (vendor.js:123902:1062)
    at Generator.<anonymous> (vendor.js:123902:3006)
    at Generator.next (vendor.js:123902:1699)
    at fulfilled (vendor.js:123914:24)
    at ZoneDelegate.invoke (zone.js:386:30)
    at Object.onInvoke (core.js:27558:33)
```

EDIT: see my comment downthread for details.

**Comment by mattStorer** [comment]
hm, I'm thinking that I misread stuff in my post above.

I believe that _priority_ is getting passed to _getConceptDisplayString_ where in that function, _priority_ is called _code_, in which case it seems like this should be working correctly.

I will work with Christina and Dave who reported the bug to see if I can look at the data they're seeing that's causing the problem.

Will report back when I get that information.

**Comment by mattStorer** [comment]
yup, okay here's the problem -

when a _Goal_ doesn't have the _priority_ element at all, this is breaking.

This code:

![image](https://github.com/user-attachments/assets/57db63a0-0da7-4450-a78a-5055e0b475e7)

with this resource:
```
{
  "resourceType": "Bundle",
  "type": "searchset",
  "total": 1,
  "link": [
    {
      "relation": "self",
      "url": "https://epicmobile.ohsu.edu/FHIRDEV/api/FHIR/R4/Goal?patient=eEgBqKAl0aFdrp.ydPAgXEWtx3g8nHroCoTHsrf9dAjc3"
    }
  ],
  "entry": [
    {
      "link": [
        {
          "relation": "self",
          "url": "https://epicmobile.ohsu.edu/FHIRDEV/api/FHIR/R4/Goal/eJn9EP0EL6VCDZVULRH4J4VHF7HF4RpMyimEMUe8rLbo3"
        }
      ],
      "fullUrl": "https://epicmobile.ohsu.edu/FHIRDEV/api/FHIR/R4/Goal/eJn9EP0EL6VCDZVULRH4J4VHF7HF4RpMyimEMUe8rLbo3",
      "resource": {
        "resourceType": "Goal",
        "id": "eJn9EP0EL6VCDZVULRH4J4VHF7HF4RpMyimEMUe8rLbo3",
        "lifecycleStatus": "active",
        "category": [
          {
            "coding": [
              {
                "system": "urn:oid:1.2.840.114350.1.13.135.3.7.4.737184.20005",
                "code": "4",
                "display": "Blood Pressure"
              }
            ],
            "text": "Blood Pressure"
          }
        ],
        "description": {
          "text": "Blood Pressure < 130/80"
        },
        "subject": {
          "reference": "Patient/eEgBqKAl0aFdrp.ydPAgXEWtx3g8nHroCoTHsrf9dAjc3",
          "display": "Fhir, Andrew"
        },
        "startDate": "2024-07-10",
        "expressedBy": {
          "reference": "Practitioner/exaRHNYJEi0wasU3QIPxPjw3",
          "type": "Practitioner",
          "display": "David Dorr, MD"
        },
        "addresses": [
          {
            "reference": "Condition/eZwj1fc34cDGuCT7al.5.YJMi1MTf19YbKlsMAU9B5aE3",
            "display": "Hypertension goal BP (blood pressure) < 130/80"
          }
        ]
      },
      "search": {
        "mode": "match"
      }
    }
  ]
}
```

_getConceptDisplayString_ needs to perform a `null` check on _code_ before trying to process _code.text_.

**Comment by mattStorer** [comment]
this is fixed in eCarePlanner 3.2.2
